### PR TITLE
Support claim attribute mappings in bulk

### PR DESCRIPTION
**Pull request by Thisara-Welmilla**
Related issue:
- https://github.com/wso2/product-is/issues/12253
- 
After following PR is merged
- https://github.com/wso2/carbon-identity-framework/pull/3647

Implementation changes
- Modify https://is.docs.wso2.com/en/latest/develop/userstore-rest-api/#/User%20Store `/userstores` API to support claim attribute mappings in bulk at user store creation.
- Introduce a new API to update the claim attribute mapping in bulk for a specific user store.

### CODE REVIEW

**Review by JKAUSHALYA — CHANGES_REQUESTED** [review]

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConfigServiceHolder.java [hunk 1]
```diff
@@ -54,4 +56,14 @@ public void setRealmService(RealmService realmService) {
 
         UserStoreConfigServiceHolder.getInstance().realmService = realmService;
     }
+
+    public ClaimMetadataManagementService getClaimMetadataManagementService() {
+
+        return UserStoreConfigServiceHolder.getInstance().claimMetadataManagementService;
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConfigServiceHolder.java:62]: As this class is a singleton, the `UserStoreConfigServiceHolder.getInstance()` refers to a static intance. therefore have implement in this way.

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConfigServiceHolder.java [hunk 2]
```diff
@@ -54,4 +56,14 @@ public void setRealmService(RealmService realmService) {
...
+    }
+
+    public void setClaimMetadataManagementService(ClaimMetadataManagementService claimMetadataManagementService) {
+
+        UserStoreConfigServiceHolder.getInstance().claimMetadataManagementService = claimMetadataManagementService;
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/UserStoreConfigServiceHolder.java:67]: As this class is a singleton, the `UserStoreConfigServiceHolder.getInstance()` refers to a static intance. therefore have implement in this way.

File: components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/factory/ClaimMetadataManagementServiceFactory.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public Class<ClaimMetadataManagementService> getObjectType() {
+
+        return ClaimMetadataManagementService.class;
+    }
+
+    @Override
+    protected ClaimMetadataManagementService createInstance() throws Exception {
+
+        if (this.claimMetadataManagementService == null) {
```
[the remark this answers is not in the document]
Comment (Thisara-Welmilla) [reply components/org.wso2.carbon.identity.api.server.userstore/org.wso2.carbon.identity.api.server.userstore.common/src/main/java/org/wso2/carbon/identity/api/server/userstore/common/factory/ClaimMetadataManagementServiceFactory.java]: Addressed with commit 734474d456011dc9caf613a1a7a0f9876f511247

**Comment by Thisara-Welmilla** [comment]
Reopened with https://github.com/wso2/identity-api-server/pull/301

**Pull request closed without merging**